sdk/ml/azure-ai-ml/.pre-commit-config.yaml

```diff
@@ -0,0 +1,13 @@
+files: sdk/ml/
+repos:
+-   repo: https://github.com/psf/black
+    rev: 22.3.0 # Needs to be updated whenever azure sdk team bumps
+    hooks:
+    -   id: black
+        args: [--config=eng/black-pyproject.toml]
+- repo: https://github.com/econchick/interrogate
+  rev: 1.5.0 
+  hooks:
+  - id: interrogate
+    args: [--fail-under=35, --ignore-module, -vv]
+    exclude: ^(sdk/ml/azure-ai-ml/tests/|sdk/ml/azure-ai-ml/azure/ai/ml/_restclient|sdk/ml/azure-ai-ml/setup.py|.*\.(yml|yaml|json|md|toml|txt|in))
```

Comment:
```suggestion
    types_or: [python]
    exclude: ^(sdk/ml/azure-ai-ml/tests/|sdk/ml/azure-ai-ml/azure/ai/ml/_restclient|sdk/ml/azure-ai-ml/setup.py
```

Reply:
It would also probably be worth configuring exclusions using `interrogate`'s configuration itself: https://interrogate.readthedocs.io/en/latest/#configuration . This would get added to our pyproject.toml

In the event we are able to get CI for interrogate, it'll also reuse the config from `pyproject.toml`.

Reply:
I tried this originally but ran into issues where configurations in the args were being accepted, but the exclude section wasn't being included when it was set in the pyproject.toml. I spent time debugging but couldn't figure out why so I just stuck it in here. 

I'll add the pre-commit config for interrogate in and just make a comment saying that the exclude part isn't being accepted due to a bug to explain it appearing here. Hopefully, I'll eventually debug it and it can clean up the style.

Reply:
You might have to explicitly point interrogate to our `pyproject.toml`

```
    args: ["--config", "sdk/ml/azure-ai-ml/pyproject.toml"
```